src/App.tsx

```diff
@@ -15,33 +14,121 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const goods = goodsFromServer;
+
+  const [isVisible, setVisible] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+  const [isSorted, setIsSorted] = useState(false);
+  const [sortBy, setSortBy] = useState('alphabetically');
+
+  const visibleGoodsList = () => setVisible(!isVisible);
+
+  const reverse = () => setIsReversed(!isReversed);
```

Comment:
I think, that if you will reverse goods in handler, it will be easier to deal with sorting problems 
BTW, I highly recommend you to read this article about events: [Handy Naming Conventions for Event Handler Functions & Props in React](https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364)

Reply:
Thank you for this article